truffle/com.oracle.truffle.sl/src/com/oracle/truffle/sl/nodes/SLStatementNode.java

```diff
@@ -88,6 +83,19 @@ public SLStatementNode getNonWrapperNode() {
         return this;
     }
 
+    public void setDebugHalt(boolean isDebugHalt) {
+        this.isDebugHalt = isDebugHalt;
+    }
+
+    @Override
+    protected boolean isTaggedWith(String tag) {
+        switch (tag) {
+            case Debugger.HALT_TAG:
```

Comment:
couldn't this be solved much simpler now, by just checking whether the parent is a sequence node?